src/components/slide-toggle/slide-toggle.ts

```diff
@@ -214,6 +240,54 @@ export class MdSlideToggle implements AfterContentInit, ControlValueAccessor {
     this._change.emit(event);
   }
 
+
+  private _onDragStart() {
+    if (this._dragPointer) {
+      return;
+    }
+
+    let thumbBarEl = this._elementRef.nativeElement.querySelector('.md-slide-toggle-bar');
+
+    this._dragPointer = {
+      barWidth: thumbBarEl.getBoundingClientRect().width
+    };
+
+    // Mark the thumb container with the `md-dragging` class to remove the duration of
+    // transform transition.
+    this._thumbContainerRef.nativeElement.classList.add('md-dragging');
+  }
+
+  private _onDrag(event: HammerInput) {
+    if (!this._dragPointer) {
+      return;
+    }
+
+    let percentage = (event.deltaX / this._dragPointer.barWidth) * 100;
+
+    // When the `slide-toggle` is currently checked, we invert the percentage, because it
+    // starts from the right.
+    percentage = this.checked ? 100 + percentage : percentage;
+    percentage = Math.max(0, Math.min(percentage, 100));
+
+    this._dragPointer.percentage = percentage;
+    this._thumbContainerRef.nativeElement.style.transform = `translate3d(${percentage}%, 0, 0)`;

```

Comment:
Use the `applyTransform` from core/style, which also sets `webkitTransform` for iOS 8